### ISSUE TITLE
Some fields have dispose_ method but it probably should be dispose

### ISSUE THREAD

**Issue by picklesrus** [LLK/scratch-blocks#1771]
While making the note picker field @ericrosenbaum noticed that dispose_ on that field was not being called. In most fields, including the super class, it is called dispose instead.
We've made this error in a few places:
field_angle
field_matrix
field_note

I also noticed this upstream in field_angle. @rachel-fenichel : is this just a typo/bug or am I missing something?

**Comment by epicfaace** [comment]
@rachel-fenichel is this something that should be fixed?

**Comment by rachel-fenichel** [comment]
Yeah this is a bug, and it looks like it got propagated through on fields that use the widget div.  Worth fixing, but low priority.
